### DISCUSSION TITLE
PhysicsBody.physics is private

### DISCUSSION THREAD

**Discussion by bleuscyther** [enable3d/enable3d#298] [Help / Questions] [answered]
Hi, I am trying to highlight (`DecalGeometry`) collision points and play a sound depending on the strength of impact (impulse?)
There must be a very good reason why `PhysicsBody.physics` is private, but is there another way to get things like the contact point (s) and getAppliedImpulse?

In one use case, I was able to manage with the velocity on impact but now it's not enough 😅.

Could anyone point me out on a good direction?

**Comment by yandeu** [comment, marked as answer]
Enable3d does only calculate the impact for breakable objects.
And there are still many hard-coded values.

See: https://github.com/enable3d/enable3d/blob/20a8e3245903e87ed8d4504077c40dc3b985f55b/packages/ammoPhysics/src/physics.ts#L354

**Reply by bleuscyther** [reply]
Oh! thank you. My bad, I skipped the  ` if (!checkCollisions && !checkBreakable) continue`  and ` if (!checkBreakable) continue`  while reading previously.

Now I understand the value of  `this.impactPoint`

**Reply by yandeu** [reply]
I guess we could change the code to make the impact details more accessible.

**Reply by yandeu** [reply]
@bleuscyther see https://github.com/enable3d/enable3d/commit/b74a3590858505e01895d6d8de68c78f5dcf1b8a

**Reply by bleuscyther** [reply]
@yandeu This was perfect, not only I am able to filter out collisions that are negligible but I can also pick the best sound effect based on the impact strength.
Thank you  🙏

**Reply by bleuscyther** [reply]
FYI, @yandeu this change is not yet available on NPM

**Reply by yandeu** [reply]
Right. It will be available once I release a new version.
Do you need it right now?

**Reply by bleuscyther** [reply]
No, don't worry. I just noticed it when deploying on a remote server. Thank you for the help tho I appreciate
